drivers/bm/bm_dfu_client.c

```diff
@@ -0,0 +1,439 @@
+#include <init.h>
+#include <logging/log.h>
+#include <storage/flash_map.h>
+#include <bootutil/bootutil_public.h>
+#include <dfu/mcuboot.h>
+#include <sys/reboot.h>
+
+#include <drivers/bm/bm_serial.h>
+#include <drivers/bm/bm_dfu.h>
+#include <drivers/bm/bm_dfu_client.h>
+#include <drivers/bm/bm_common.h>
+
+LOG_MODULE_REGISTER(bm_dfu_client, CONFIG_BM_LOG_LEVEL);
+
+static dfu_client_ctx_t _client_context;
+
+static void bm_dfu_client_req_next_chunk(void)
+{
+    bm_frame_header_t frm_hdr;
+    bm_frame_t *chunk_req_frm;
+    bm_dfu_event_chunk_request_t chunk_req_evt;
+    uint8_t tx_buf[sizeof(bm_frame_header_t) + sizeof(bm_dfu_event_chunk_request_t) + sizeof(bm_dfu_frame_header_t)];
```

Comment:
How big is this normally? If it's too big, I'm worried about allocating it all in the stack.

Reply:
I think the max size for this would be roughly 260-270 bytes